app/models/mixins/authentication_mixin.rb

```diff
@@ -149,6 +149,11 @@ def missing_credentials?(type = nil)
     !has_credentials?(type)
   end
 
+  def provider_authentication_status_ok?(type = nil)
+    authtype = [type, default_authentication_type].compact
+    authentication_for_providers.find_by(:authtype => authtype).try(:status) == "Valid"
```

Comment:
Minor, but I think this is passing an array to find_by, which is like doing an IN query and I guess getting the first one?  If so, I think you need an ORDER clause or something to make sure it's deterministic.

Reply:
Yeah actually want to do more than order, we need to prioritize the requested type if it is passed in then fall back to the default type.

Added specs to cover the cases I think we want to support